src/main/java/seedu/toluist/controller/TagController.java

```diff
@@ -31,95 +27,78 @@
     private static final int SECTION_INDEX = 0;
     private static final int SECTION_KEYWORDS = 1;
 
-    private static final String MESSAGE_TEMPLATE_SUCCESS = "Successfully added \"%s\".\n";
-    private static final String MESSAGE_TEMPLATE_FAIL = "Failed to add \"%s\".\n";
-    private static final String MESSAGE_TEMPLATE_RESULT = "%s%s successfully added.";
-
-    private static final String HELP_DETAILS = "Adds a tag(s) to an existing task.";
-    private static final String HELP_FORMAT = "tag INDEX TAG(S)";
-    private static final String[] HELP_COMMENTS = { "Related commands: `untag`",
-                                                    "All tags are one word long.",
-                                                    "Each word entered after the index will be its own tag.", };
-    private static final String[] HELP_EXAMPLES = { "`tag 1 schoolwork`\n"
-                                                        + "Adds the tag `schoolwork` to the task at index 1.",
-                                                    "`tag 1 housework groceries`\nAdds the tags "
-                                                        + "`housework` and `groceries` to the task at index 1." };
-
-    private static final Logger logger = LogsCenter.getLogger(TagController.class);
-
     public void execute(Map<String, String> tokens) throws InvalidCommandException {
-        logger.info(getClass() + "will handle command");
-
-        // initialize keywords and variables for searching
-        String[] keywordList = StringUtil.convertToArray(tokens.get(PARAMETER_KEYWORDS));
-        int index = Integer.parseInt(tokens.get(PARAMETER_INDEX)) - 1;
-        TodoList todoList = TodoList.getInstance();
-        Task task = UiStore.getInstance().getShownTasks().get(index);
-        ArrayList<String> successfulList = new ArrayList<>();
-        ArrayList<String> failedList = new ArrayList<>();
-
-        for (String keyword : keywordList) {
-            if (task.addTag(new Tag(keyword))) {
-                successfulList.add(keyword);
-            } else {
-                failedList.add(keyword);
-            }
+        if (isInvalidFormat(tokens)) {
+            showInvalidFormatMessage();
```

Comment:
also can actually combine the if () show to just `validateValidFormat()`. Similarly for the bottom, can combine if (isIndex... ) { } to `validateIndexNotOutOfBounds`